client/src/components/JurisdictionAdmin/BatchInventory.tsx

```diff
@@ -0,0 +1,502 @@
+import {
+  useQueryClient,
+  useMutation,
+  useQuery,
+  UseQueryResult,
+  UseMutationResult,
+} from 'react-query'
+import styled from 'styled-components'
+import { useParams, Link } from 'react-router-dom'
+import { Wrapper, Inner } from '../Atoms/Wrapper'
+import FileUpload from '../Atoms/FileUpload'
+import { fetchApi } from '../../utils/api'
+import AsyncButton from '../Atoms/AsyncButton'
+import { FileProcessingStatus } from '../useCSV'
+import {
+  IFileUpload,
+  useUploadFiles,
+  useDeleteFile,
+  useUploadedFile,
+} from '../useFileUpload'
+import { apiDownload } from '../utilities'
+import LinkButton from '../Atoms/LinkButton'
+
+const STEPS = [
+  'Upload Election Results',
+  'Inventory Batches',
+  'Download Audit Files',
+] as const
+
+const isUploaded = (fileUpload: IFileUpload) =>
+  fileUpload.uploadedFile.data?.processing?.status ===
+  FileProcessingStatus.PROCESSED
+
+const useBatchInventoryCVR = (
+  electionId: string,
+  jurisdictionId: string
+): IFileUpload => {
+  const key = ['batchInventory', jurisdictionId, 'cvr']
+  const url = `/api/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/cvr`
+  const uploadFiles = useUploadFiles(key, url)
+  const deleteFile = useDeleteFile(key, url)
+  const queryClient = useQueryClient()
+  return {
+    uploadedFile: useUploadedFile(key, url, {
+      onFileChange: () =>
+        // Tabulator status file is reprocessed when CVR is uploaded
+        queryClient.invalidateQueries([
+          'batchInventory',
+          jurisdictionId,
+          'tabulatorStatus',
+        ]),
+    }),
+    uploadFiles: files => {
+      const formData = new FormData()
+      formData.append('cvr', files[0], files[0].name)
+      return uploadFiles.mutateAsync(formData)
+    },
+    uploadProgress: uploadFiles.progress,
+    deleteFile: () => deleteFile.mutateAsync(),
+    // TODO make this actually work
+    downloadFileUrl: `${url}/file`,
+  }
+}
+
+const useBatchInventoryTabulatorStatus = (
+  electionId: string,
+  jurisdictionId: string
+): IFileUpload => {
+  const key = ['batchInventory', jurisdictionId, 'tabulatorStatus']
+  const url = `/api/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/tabulator-status`
+  const uploadFiles = useUploadFiles(key, url)
+  const deleteFile = useDeleteFile(key, url)
+  return {
+    uploadedFile: useUploadedFile(key, url),
+    uploadFiles: files => {
+      const formData = new FormData()
+      formData.append('tabulatorStatus', files[0], files[0].name)
+      return uploadFiles.mutateAsync(formData)
+    },
+    uploadProgress: uploadFiles.progress,
+    deleteFile: () => deleteFile.mutateAsync(),
+    // TODO make this actually work
+    downloadFileUrl: `${url}/file`,
+  }
+}
+
+interface ISignOffQueries {
+  status: UseQueryResult<{ signedOffAt: string | null }>
+  signOff: UseMutationResult<unknown, unknown, void, unknown>
+  undoSignOff: UseMutationResult<unknown, unknown, void, unknown>
```

Comment:
Very minor, but the first param to these `UseMutationResult` generics could be `void` instead of `unknown`, since the mutation functions (`() => fetchApi(url, { method: 'POST' })` and `() => fetchApi(url, { method: 'DELETE' })`) both return `Promise<void>`:

```ts
signOff: UseMutationResult<void, unknown, void, unknown>
undoSignOff: UseMutationResult<void, unknown, void, unknown>
```

Reply:
Nice, thanks! cf37105b